controllers/configurationpolicy_controller.go

```diff
@@ -297,6 +299,13 @@ func (r *ConfigurationPolicyReconciler) refreshDiscoveryInfo() error {
 
 	dd := r.TargetK8sClient.Discovery()
 
+	serverVersion, err := dd.ServerVersion()
+	if err != nil {
+		log.Error(err, "Could not get the server version")
+	}
+
+	r.serverVersion = serverVersion.String()
```

Comment:
Will this work if `err != nil`?

Reply:
Assuming that in that case it is just set to an empty string, then the semver library would treat it as invalid and do this:
> An invalid semantic version string is considered less than a valid one. All invalid semantic version strings compare equal to each other.

So I think it would fallback to the "old" validation gracefully.

Reply:
Yes. @JustinKuli is correct. in that case it would fallback to the original validation.